truffle/src/main/java/org/jruby/truffle/nodes/core/KernelNodes.java

```diff
@@ -1562,7 +1567,12 @@ public boolean doesRespondTo(VirtualFrame frame, Object object, RubyString name,
 
         @Specialization
         public boolean doesRespondTo(VirtualFrame frame, Object object, RubySymbol name, UndefinedPlaceholder checkVisibility) {
-            return dispatch.doesRespondTo(frame, name, object);
+            return doesRespondTo(frame, object, name, false);
+        }
+
+        @Specialization
+        public boolean doesRespondTo(VirtualFrame frame, Object object, RubySymbol name, RubyBasicObject checkVisibility) {
+            return doesRespondTo(frame, object, name, false);
```

Comment:
It would work for the name but if you want to remove the UndefinedPlaceholder, BooleanCastNode has to handle it somehow.